### ISSUE TITLE
[BUG]: 403 Forbidden with homemade Android build

### ISSUE THREAD

**Issue by dic1911** [OneKeyHQ/app-monorepo#5584]
**Prerequisites**
Please answer the following questions for yourself before submitting an issue. **YOU MAY DELETE THE PREREQUISITES SECTION.**

- [x] I checked to make sure that this issue has not already been filed
- I've asked for help in the [OneKey Discord](https://discord.com/invite/onekey) before filing this issue.

^ Discord is closed and you should know it. Please update the template.

**Describe the bug**
403 Forbidden keep popping up, unable to show wallet balance and probably everything that depends on your API properly.

**To Reproduce**
1. Build app from source for Android, install and run
2. Set password as asked, wait for migration
3. Be ready for a broken wallet

**Expected behavior**
Should there be no 403 forbidden errors, and everything should be displayed properly, as it did for older releases.

**Screenshots**
![image](https://github.com/user-attachments/assets/90b79a2d-823c-4e2d-aa41-ed4ba2fee0bc)

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**
 - Device: Pixel 4a 5G
 - OS: Android 14
 - Browser: N/A (not applicable as I'm talking about native app here)
 - Version: N/A

**Additional context**

**Comment by ByteZhang1024** [comment]
The error is not available in your area.

**Comment by dic1911** [comment]
> The error is not available in your area.

@ByteZhang1024 Dude, you're not getting it, the point is I could switch to an older version of the app and use everything just fine, please read my issue again and don't skip any part of it.

For further information, I had desktop and web built in same environment and things work just fine, no 403 errors.

**Comment by ByteZhang1024** [comment]
Are your mobile device, web, and desktop versions operating in the same network environment?

According to laws and regulations in different countries and regions, we have implemented corresponding restrictions on our API.

**Comment by ByteZhang1024** [comment]
Old versions fully utilized public RPC services, not our own services, so there were no restrictions.

**Comment by dic1911** [comment]
> Are your mobile device, web, and desktop versions operating in the same network environment?
> 
> According to laws and regulations in different countries and regions, we have implemented corresponding restrictions on our API.

Yeah, everything is under the same network environment, and the title wouldn't be like this if it doesn't work for me on both PC and Android.

**Comment by dic1911** [comment]
One other thing, I live in a place where you ship the wallet directly to, how is that acceptable that you guys simply decided to cripple user experience after taking my money?

**Comment by ByteZhang1024** [comment]
Can you tell us your country and region, so our colleagues can check if there are any issues?

**Comment by dic1911** [comment]
@ByteZhang1024 Can we talk in either email or somewhere that's not public instead?

**Comment by ByteZhang1024** [comment]
Of course, the work order can only be seen internally.
https://help.onekey.so/hc/requests/new

**Comment by dic1911** [comment]
Ticket opened.
![image](https://github.com/user-attachments/assets/cfd4deeb-d15b-44a3-b8f3-f53521c0a001)

**Comment by ByteZhang1024** [comment]
We've seen it. We'll investigate the issue, and if we need more help from you, I'll reply to you here.

**Comment by ByteZhang1024** [comment]
If it's convenient, could you provide an IP address? Based on our investigation so far, it appears your request was blocked by the firewall.

Just use a  https://help.onekey.so/hc/requests/new

**Comment by dic1911** [comment]
@ByteZhang1024 just had another ticket opened, ID is 30853

**Comment by ByteZhang1024** [comment]
Relevant colleagues have given me feedback. 
We would like to know your purpose for using the API to develop your own app. If you wish to continue with secondary development, our server has some restrictions for non-OneKey official wallets. 
If you want to open up these permissions, you can contact our business team for assistance with integration.

**Comment by dic1911** [comment]
@ByteZhang1024 What's the meaning of open source to your team then?
Marketing with open source and pretend to be good guys while locking up the APIs so only you and your team members could have real use of the source code? You must be joking.

**Comment by ByteZhang1024** [comment]
The purpose of open source is to prove that the software released by OneKey does not maliciously collect users' mnemonics and is safe, reliable, and trustworthy. We also welcome everyone to act as white hat hackers to discover vulnerabilities and submit them to us.

**Comment by dic1911** [comment]
@ByteZhang1024 That doesn't make any sense, if only builds from your side works, how could outsiders like me verify the authenticity of your binary release?

**Comment by ByteZhang1024** [comment]
Desktop, Extension, and Web versions are directly packaged using GitHub CI, and you can see all the corresponding files. You can go and verify them directly.
For mobile versions, verification might be a bit troublesome at the moment.

**Comment by dic1911** [comment]
@ByteZhang1024 And the troublesome part is exactly what I'm asking and the reason why I'm here.

If you guys decide to be so hostile toward open source, please kindly do the community a favor, don't put the open source tag on yourselves, you don't deserve it with such attitude.

**Comment by luchenqun** [comment]
> Old versions fully utilized public RPC services, not our own services, so there were no restrictions.

@ByteZhang1024 Sorry to bother you. Thank you for the open-source version, which has laid a very good foundation for wallet development. Here's the thing: we also want to develop a wallet based on your version, and after using the latest version, we've found the problems mentioned above. I'd like to know from which version you started using your self-built services. I'm wondering whether the last version of 4.x, which is 4.24.0, is still using the public RPC services? thank you.

**Comment by ByteZhang1024** [comment]
@luchenqun 
4.x is using a public PRC.
5.x uses a self-built RPC by Onekey, but there is also an option to use a custom RPC.

**Comment by luchenqun** [comment]
@ByteZhang1024 

Thank you for your answer. I'm not sure if I have clearly described my question.

Generally speaking, I hope that when I select a tag that is as new as possible and recompile it, the error "403 forbidden, our services are not available in your region" will not occur. Because you mentioned above: "Old versions fully utilized public RPC services, not our own services, so there were no restrictions."

In your answer, you mentioned that "5.x uses a self-built RPC by Onekey, but there is also an option to use a custom RPC." I'd like to know whether all services in 5.x can use public RPC services to avoid the "403 forbidden" error. If so, could you briefly introduce how to configure it?

If a lot of code need to be modified in 5.x to prevent the occurrence of the "403 forbidden" error, can I use v4.20.0 to ensure that the "403 forbidden" error will definitely not occur? 

Thank you very much.
